src/Algebra/Graph/Relation/Symmetric.hs

```diff
@@ -13,25 +13,509 @@ module Algebra.Graph.Relation.Symmetric (
     -- * Data structure
     SymmetricRelation, fromRelation, toRelation,
 
+    -- * Basic graph construction primitives
+    empty, vertex, edge, overlay, connect, vertices, edges, overlays, connects,
+
+    -- * Relations on graphs
+    isSubgraphOf,
+
     -- * Graph properties
-    neighbours
+    isEmpty, hasVertex, hasEdge, vertexCount, edgeCount, vertexList, edgeList,
+    adjacencyList, vertexSet, edgeSet, preSet, postSet, neighbours,
+
+    -- * Standard families of graphs
+    path, circuit, clique, biclique, star, stars, tree, forest,
+
+    -- * Graph transformation
+    removeVertex, removeEdge, replaceVertex, mergeVertices, transpose, gmap, induce,
+
+    -- * Relational operations
+    compose, -- closure, reflexiveClosure, symmetricClosure, transitiveClosure
   ) where
 
-import Algebra.Graph.Relation
+import qualified Algebra.Graph.Relation as R
+import qualified Algebra.Graph.Relation.Internal as RI
 import Algebra.Graph.Relation.InternalDerived
+import Algebra.Graph.Class (connect, empty, vertex, overlay)
 
 import qualified Data.Set as Set
+import Data.Tuple
+import Data.Tree 
+import qualified Data.Tree as Tree
 
 -- | Construct a symmetric relation from a 'Relation'.
 -- Complexity: /O(1)/ time.
-fromRelation :: Relation a -> SymmetricRelation a
+fromRelation :: R.Relation a -> SymmetricRelation a
 fromRelation = SymmetricRelation
 
 -- | Extract the underlying relation.
 -- Complexity: /O(m*log(m))/ time.
-toRelation :: Ord a => SymmetricRelation a -> Relation a
-toRelation = symmetricClosure . fromSymmetric
+toRelation :: Ord a => SymmetricRelation a -> R.Relation a
+toRelation = R.symmetricClosure . fromSymmetric
+
+-- | Construct the graph comprising /a single edge/.
+-- Complexity: /O(1)/ time, memory and size.
+--
+-- @
+-- edge x y               == 'overlay' ('connect' ('vertex' x) ('vertex'
+-- y)) ('connect' ('vertex' y) ('vertex' x))
```

Comment:
Make sure the Haddock is generated correctly -- this line needs to be fixed, for example.

Reply:
What's wrong here? the identation?

Reply:
This is still not resolved: what's up with this comment: `-- y)) ('connect' ('vertex' y) ('vertex' x))` ?

Reply:
Sorry I messed up the comments! I hope I have it fixed now